src/engine/ItemmarkManager.ts

```diff
@@ -0,0 +1,105 @@
+import type { IMediaContainer } from './providers/MediaPlugin';
+import { type StorageController, Store } from './StorageController';
+
+export class ItemmarkManager {
+
+    private readonly Items:Map<string,ItemMark[]>=new Map();
+    constructor(private readonly storage: StorageController) {
+        this.Load();
+    }
+
+    private async Load() {
+        const mediamarks = await this.storage.LoadPersistent<ItemmarkSerialized[]>(Store.Itemmarks);
```

Comment:
Can't this be loaded directly without (de-)serialization? `this.Items = await this.storage.LoadPersistent<ItemmarkSerialized[]>(Store.Itemmarks);`

Reply:
If we save as map directly, it would be Loaded back as a map ?

Reply:
If it doesn't work use a `Record<string, MarkItem[]>` instead

Reply:
> If we save as map directly, it would be Loaded back as a map ?

Good catch, this may solv this if keeping `Map`
```javascript
const map = new Map([ [ 'a', 1 ], [ 'b', 2 ] ]);
const serialized = [ ...map.entries() ];
const deserialized = new Map(serialized);
```